tests/reporter/kubernetes.go

```diff
@@ -173,6 +181,133 @@ func (r *KubernetesReporter) logVMIs(virtCli kubecli.KubevirtClient) {
 	fmt.Fprintln(f, string(j))
 }
 
+func (r *KubernetesReporter) logDMESG(virtCli kubecli.KubevirtClient, since time.Time) {
+
+	logsdir := filepath.Join(r.artifactsDir, "nodes")
+
+	if err := os.MkdirAll(logsdir, 0777); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create directory: %v\n", err)
+		return
+	}
+
+	vmis, err := virtCli.VirtualMachineInstance(v1.NamespaceAll).List(&v12.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch vmis: %v\n", err)
+		return
+	}
+
+	nodes := map[string]struct{}{}
+	for _, vmi := range vmis.Items {
+		if vmi.Status.NodeName != "" {
+			nodes[vmi.Status.NodeName] = struct{}{}
+		}
+	}
+
+	rex := regexp.MustCompile(`\[([^]]+)]`)
+	for node, _ := range nodes {
+		func() {
```

Comment:
That's because of the `defer`.

Reply:
Oh, I didn't notice it. Also, since I mentioned it, I _think_ this would this work if func(s) were separate goroutines (not that we are interested in performance here)